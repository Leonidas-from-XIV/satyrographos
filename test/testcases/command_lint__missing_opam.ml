open Core

let opam_file_without_name_field =
  "satysfi-package-doc.opam", sprintf
    {|opam-version: "2.0"
|}

let satyristes =
  "Satyristes", sprintf
    {|(version "0.0.2")

(library
  (name "package")
  (version "0.1")
  (sources ())
  (opam "satysfi-package.opam")
  (dependencies ()))

(libraryDoc
  (name "package-doc")
  (version "0.1")
  (build ())
  (sources ())
  (opam "satysfi-package-doc.opam")
  (dependencies ((package ()))))
|}

let files =
  [ opam_file_without_name_field;
    satyristes;
  ]

let () =
  TestCommand.test_lint_command files
