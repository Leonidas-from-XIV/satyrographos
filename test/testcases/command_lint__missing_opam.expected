Exception:
OpamSystem.Internal_error("File @@test_library@@/satysfi-package.opam does not exist or can't be read")
